### PR TITLE
monitor: init at 0.6.1

### PR DESCRIPTION
**Pull request by xiorcale**
###### Motivation for this change

Add more curated apps from Elementary OS app store for Pantheon Desktop.

###### Things done

 Everything seems to work fine, except for the Wingpanel indicator. As I've understood, there should be one (as far as the option is checked), but unfortunately nothing appears for me. 

Note that I haven't tested the app on ElementaryOS to see what the indicator should look like.

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/applications/misc/monitor/default.nix [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    gtk3
+    pantheon.granite
+    pantheon.wingpanel
+    libgee
+    libgtop
+    libwnck3
+    libxml2
+  ];
+
+  PKG_CONFIG_WINGPANEL_2_0_INDICATORSDIR = "${placeholder "out"}/lib/wingpanel";
```
Comment (worldofpeace) [inline pkgs/applications/misc/monitor/default.nix]:
```suggestion
```

I've created this PR to eliminate this issue https://github.com/stsdc/monitor/pull/141. Can you fetch the patches?

File: pkgs/applications/misc/monitor/default.nix [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  buildInputs = [
+    bamf
+    glib
+    gtk3
+    pantheon.granite
+    pantheon.wingpanel
+    libgee
+    libgtop
+    libwnck3
+    libxml2
```
Comment (worldofpeace) [inline pkgs/applications/misc/monitor/default.nix]:
```suggestion
```

Where is this needed? It builds without it.

File: pkgs/applications/misc/monitor/default.nix [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+, python3
+, gettext
+, glib
+, gtk3
+, bamf
+, libwnck3
+, libgee
+, libgtop
+, libxml2
+, wrapGAppsHook }:
```
Comment (worldofpeace) [inline pkgs/applications/misc/monitor/default.nix]:
```suggestion
, wrapGAppsHook 
}:
```

File: pkgs/applications/misc/monitor/default.nix [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    rev = version;
+    sha256 = "17z1m193s7qygavfwd8qsw97blxbfmq9gnsymdjlc1ddk8hldw0z";
+  };
+
+  nativeBuildInputs = [
+    gettext
+    meson
+    ninja
+    vala
+    pkgconfig
```
Comment (worldofpeace) [inline pkgs/applications/misc/monitor/default.nix]:
```suggestion
    pkg-config
```

This is an alias yet to be deprecated

File: pkgs/applications/misc/monitor/default.nix (hunk 2 above)
Comment (kjuvi) [reply pkgs/applications/misc/monitor/default.nix]: Meson doesn't have it in its dependencies but the README lists it as a dependency, which confused me. I'll remove it.

File: pkgs/applications/misc/monitor/default.nix (hunk 4 above)
Comment (kjuvi) [reply pkgs/applications/misc/monitor/default.nix]: Good to know, thank's

File: pkgs/applications/misc/monitor/default.nix (hunk 1 above)
Comment (kjuvi) [reply pkgs/applications/misc/monitor/default.nix]: Building with your patch did eliminate the issue, but I still can't see the the indicator in Wingpanel :sweat_smile:

**Comment by worldofpeace** [comment]
> Building with your patch did eliminate the issue, but I still can't see the the indicator in Wingpanel sweat_smile

I've patched wingpanel for purity, so wingpanel-with-indicators takes a `indicators` argument so you can configure it.

```
with import ./. {};

pantheon.wingpanel-with-indicators.override {
  indicators = [
    monitor
  ] ++ pantheon.wingpanelIndicators;
}
```

I specifically didn't want it so it reads indicators from `/run/current-system/sw/*` (it makes things harder).

But I haven't added a nixos interface to this because there was no package extra package in nixos that had an extra indicator. I should do this now :+1: .

**Comment by worldofpeace** [comment]
@kjuvi I don't think the directory structure here is best.
Can we add an `applications/system` like `tools/system`? What do you think?

**Comment by kjuvi** [comment]
> @kjuvi I don't think the directory structure here is best.
> Can we add an `applications/system` like `tools/system`? What do you think?

I was actually looking for this specific folder but since it didn't exist I just put it in applications/misc by default.

I'll add this folder and move the expression there :+1:

**Comment by worldofpeace** [comment]
@kjuvi It seems you've changed the package version to the rev of my patches. Can you return to https://github.com/stsdc/monitor/releases/tag/0.6.1, and just `fetchpatch` https://github.com/stsdc/monitor/commit/07d1984175fcaef2909029a387f830efd647471b.patch and https://github.com/stsdc/monitor/commit/ab2cfed150cd2a6b5c3fcee5297a65c1b429c674.patch. We should be using a released version.

**Comment by kjuvi** [comment]
> I've patched wingpanel for purity, so wingpanel-with-indicators takes a `indicators` argument so you can configure it.
> 
> ```
> with import ./. {};
> 
> pantheon.wingpanel-with-indicators.override {
>   indicators = [
>     monitor
>   ] ++ pantheon.wingpanelIndicators;
> }
> ```
> 
> I specifically didn't want it so it reads indicators from `/run/current-system/sw/*` (it makes things harder).
> 
> But I haven't added a nixos interface to this because there was no package extra package in nixos that had an extra indicator. I should do this now +1 .

Okay so, if I understood correctly, it is the responsibility of the user to override his wingpanel if he wants the indicator ?

**Comment by kjuvi** [comment]
> @kjuvi It seems you've changed the package version to the rev of my patches. Can you return to https://github.com/stsdc/monitor/releases/tag/0.6.1, and just `fetchpatch` https://github.com/stsdc/monitor/commit/07d1984175fcaef2909029a387f830efd647471b.patch and https://github.com/stsdc/monitor/commit/ab2cfed150cd2a6b5c3fcee5297a65c1b429c674.patch. We should be using a released version.

Woops, you're right, I'll revert this...

**Comment by worldofpeace** [comment]
@kjuvi I've added options to configure wingpanel and switchboard plugins #76116. 

But you can still build
```
with import ./. {};

pantheon.wingpanel-with-indicators.override {
  indicators = [ monitor ];
};
```

Note without the prior PR, it will be the only indicator in wingpanel.

**Comment by worldofpeace** [comment]
Looks like there was a little issue with that PR after I merged it https://github.com/NixOS/nixpkgs/commit/776893e4d693ffa9e8dc9e94fb14afcbeab2a7e1 :smile: 
I tested the indicator, it works LGTM.

In the force push here, I added the `updateScript` attribute for https://github.com/NixOS/nixpkgs/tree/master/pkgs/desktops/pantheon/nixpkgs_github_update.

**Pull request merged**